src/main/drivers/serial_uart_stm32f10x.c

```diff
@@ -37,22 +38,158 @@
+#else
+# define UART1_RX_DMA_CHANNEL 0
+#endif
+
+#ifdef USE_UART1_TX_DMA
+# define UART1_TX_DMA_CHANNEL DMA1_Channel4
+#else
+# define UART1_TX_DMA_CHANNEL 0
+#endif
+
+#define UART2_RX_DMA_CHANNEL 0
+#define UART2_TX_DMA_CHANNEL 0
+#define UART3_RX_DMA_CHANNEL 0
+#define UART3_TX_DMA_CHANNEL 0
+
+static uartDevice_t uartHardware[] = {
+    {
+        .dev = USART1,
+        .rxDMAChannel = UART1_RX_DMA_CHANNEL,
+        .txDMAChannel = UART1_TX_DMA_CHANNEL,
+        .pinPair = {
+            { DEFIO_TAG_E(PA10), DEFIO_TAG_E(PA9) },
+            { DEFIO_TAG_E(PB7), DEFIO_TAG_E(PB6) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        //.af = GPIO_AF_USART1,
+        .rcc = RCC_APB2(USART1),
+        .irqn = USART1_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART1_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART1
+    },
+    {
+        .dev = USART2,
+        .rxDMAChannel = UART2_RX_DMA_CHANNEL,
+        .txDMAChannel = UART2_TX_DMA_CHANNEL,
+        .pinPair = {
+            { DEFIO_TAG_E(PA3), DEFIO_TAG_E(PA2) },
+            { DEFIO_TAG_E(PD6), DEFIO_TAG_E(PD5) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        //.af = GPIO_AF_USART2,
+        .rcc = RCC_APB1(USART2),
+        .irqn = USART2_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART2,
+        .rxPriority = NVIC_PRIO_SERIALUART2
+    },
+#ifdef USE_UART3 // Limitation by FLASH & RAM (Operation not tested)
+    {
+        .dev = USART3,
+        .rxDMAChannel = UART3_RX_DMA_CHANNEL,
+        .txDMAChannel = UART3_TX_DMA_CHANNEL,
+        .pinPair = {
+            { DEFIO_TAG_E(PB11), DEFIO_TAG_E(PB10) },
+            { DEFIO_TAG_E(PD9), DEFIO_TAG_E(PD8) },
+            { DEFIO_TAG_E(PC11), DEFIO_TAG_E(PC10) },
+        },
+        .rx = IO_TAG_NONE,
+        .tx = IO_TAG_NONE,
+        //.af = GPIO_AF_USART3,
+        .rcc = RCC_APB1(USART3),
+        .irqn = USART3_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART3,
+        .rxPriority = NVIC_PRIO_SERIALUART3
+    },
+#endif
+};
+
+static uartDevice_t *uartHardwareMap[ARRAYLEN(uartHardware)];
+
+static volatile uint8_t rxBuffers[ARRAYLEN(uartHardware)][UART_RX_BUFFER_SIZE];
+static volatile uint8_t txBuffers[ARRAYLEN(uartHardware)][UART_TX_BUFFER_SIZE];
+
 void serialInitHardwareMap(serialPinConfig_t *pSerialPinConfig)
 {
-    // F1 isn't converted yet.
-    UNUSED(pSerialPinConfig);
+    memset(uartHardwareMap, 0, sizeof(uartHardwareMap));
+
+    for (size_t index = 0 ; index < ARRAYLEN(uartHardware) ; index++) {
+        uartDevice_t *uartDev = &uartHardware[index];
+        for (int pair = 0 ; pair < UART_PINPAIR_COUNT ; pair++) {
+
+            if (uartDev->pinPair[pair].rx == pSerialPinConfig->ioTagRx[index]
+                    && uartDev->pinPair[pair].tx == pSerialPinConfig->ioTagTx[index]) {
+                uartDev->rx = uartDev->pinPair[pair].rx;
+                uartDev->tx = uartDev->pinPair[pair].tx;
+                uartDev->rxBuffer = rxBuffers[index];
```

Comment:
You can probably set `uartDev->port.rxBuffer` and uartDev->port.rxBufferSize` directly here